pkg/entities/entity.go

```diff
@@ -146,62 +152,86 @@ func (e *Entities) GetEntity(guid string) (*Entity, error) {
 	return resp.Actor.Entity, nil
 }
 
+func buildEntitiesSearchQuery(resultsFields *[]string) string {
+	if resultsFields == nil {
+		return searchEntitiesQuery
+	}
+
+	f := *resultsFields
+	fields := strings.Join(f[:], "\n")
+
+	return fmt.Sprintf(`
+		query($queryBuilder: EntitySearchQueryBuilder, $cursor: String) {
+			actor {
+				entitySearch(queryBuilder: $queryBuilder) {
+					results(cursor: $cursor) {
+						nextCursor
+						entities {
+							%s
```

Comment:
This type of filtering won't work for things that are on an interface, like `ApplicationID` for APM is (see `searchEntitiesQuery`